ambry-store/src/test/java/com.github.ambry.store/CuratedLogIndexState.java

```diff
@@ -393,6 +474,119 @@ void verifyEntriesForHardDeletes(Set<MockId> deletedIds) throws IOException {
+   * @return {@code true} if the {@code id} is considered deleted at {@code referenceTimeMs}. {@code false} otherwise.
+   */
+  boolean isDeletedAt(MockId id, long referenceTimeMs) {
+    Offset deleteIndexSegmentStartOffset = indexSegmentStartOffsets.get(id).getSecond();
+    return deleteIndexSegmentStartOffset != null
+        && lastModifiedTimesInSecs.get(deleteIndexSegmentStartOffset) * Time.MsPerSec < referenceTimeMs;
+  }
+
+  /**
+   * Returns {@code true} if the {@code id} is considered expired at {@code referenceTimeMs}.
+   * @param id the {@link MockId} to check.
+   * @param referenceTimeMs the reference time in ms at which the check needs to be performed.
+   * @return {@code true} if the {@code id} is considered expired at {@code referenceTimeMs}. {@code false} otherwise.
+   */
+  boolean isExpiredAt(MockId id, long referenceTimeMs) {
+    long expiresAtMs = allKeys.get(id).getFirst().getExpiresAtMs();
+    return expiresAtMs != Utils.Infinite_Time && expiresAtMs < referenceTimeMs;
+  }
+
+  /**
+   * Ensures that the {@link PersistentIndex} is sane and correct by checking that
+   * 1. It contains no duplicate entries.
+   * 2. The ordering of PUT and DELETE entries is correct.
+   * 3. There are no offsets in the log not accounted for in the index.
+   * @throws IOException
+   * @throws StoreException
+   */
+  void verifyRealIndexSanity() throws IOException, StoreException {
+    Map<MockId, Boolean> keyToDeleteSeenMap = new HashMap<>();
+    IndexSegment prevIndexSegment = null;
+    for (IndexSegment indexSegment : index.getIndexSegments().values()) {
+      Offset indexSegmentStartOffset = indexSegment.getStartOffset();
+      if (prevIndexSegment == null) {
+        assertEquals("There are offsets in the log not accounted for in index", log.getFirstSegment().getStartOffset(),
+            indexSegmentStartOffset.getOffset());
+      } else if (prevIndexSegment.getLogSegmentName().equals(indexSegment.getLogSegmentName())) {
+        assertEquals("There are offsets in the log not accounted for in index", prevIndexSegment.getEndOffset(),
+            indexSegmentStartOffset);
+      } else {
+        LogSegment segment = log.getSegment(prevIndexSegment.getLogSegmentName());
+        assertEquals("There are offsets in the log not accounted for in index", segment.getEndOffset(),
+            prevIndexSegment.getEndOffset().getOffset());
+        segment = log.getNextSegment(segment);
+        assertEquals("There are offsets in the log not accounted for in index", segment.getStartOffset(),
+            indexSegmentStartOffset.getOffset());
+      }
+      NavigableSet<IndexEntry> indexEntries = new TreeSet<>(PersistentIndex.INDEX_ENTRIES_COMPARATOR);
+      List<MessageInfo> infos = new ArrayList<>();
+      FindEntriesCondition condition = new FindEntriesCondition(Long.MAX_VALUE);
+      indexSegment.getEntriesSince(null, condition, infos, new AtomicLong(0));
```

Comment:
just that I would like to keep it consistent. I remember elsewhere for a similar FindEntriesCondition, we did the same, and hence.